src/services/configService.js

```diff
@@ -69,4 +70,22 @@ export class ConfigService {
     return HttpClient.put(`${ENDPOINT}solution-settings/logo`, logo, options)
       .map(prepareLogoResponse);
   }
+
+  /* get solution settings */
+  static getSolutionSettings() {
+    return HttpClient
+      .get(`${ENDPOINT}solution-settings/theme`)
```

Comment:
Something I missed yesterday, the `getAzureMapKey()` is also making the call to `solution-settings/theme` so you can reuse it with a little modification. You probably need to update the model to pull out any other data you need,

```
static getSolutionSettings() {
  return HttpClient.get(`${ENDPOINT}solution-settings/theme`)
    .map(toSolutionSettingThemeModel)
}

// Reuse the getSolutionSettings() in the key call
// TODO: Save the solution settings data in the redux store to avoid making the same call twice
static getAzureMapKey() {
  return ConfigService.getSolutionSettings()
    .map(response => response.azureMapsKey);
  }
```

Reply:
got rid of getAzureMapKey now